### PR TITLE
Translate library examples to Vala

### PR DESCRIPTION
**Pull request by benpocalypse**
There are some shortcomings due to the Vala API not having access to the `application` class, but I think these translations are otherwise fairly faithful to the originals. Feel free to comment and criticize though.

### CODE REVIEW

File: src/Library/demos/Toasts/main.vala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    toast.dismissed.connect (() => {
+      button_simple.set_sensitive (true);
+    });
+    overlay.add_toast (toast);
+    button_simple.set_sensitive (false);
+  });
+
+  var button_advanced = workbench.builder.get_object("button_advanced") as Gtk.Button;
+  button_advanced.clicked.connect(() => {
+    var message_id = "42";
```
Comment (lw64) [inline src/Library/demos/Toasts/main.vala:19]: please use ```int``` instead of ```var``` here.

File: src/Library/demos/Toasts/main.vala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  var overlay = workbench.builder.get_object ("overlay") as Adw.ToastOverlay;
+
+  var button_simple = workbench.builder.get_object ("button_simple") as Gtk.Button;
+  button_simple.clicked.connect (() => {
+    var toast = new Adw.Toast ("Toasts are delicious!") {
+      timeout = 1
+    };
+
+    toast.dismissed.connect (() => {
+      button_simple.set_sensitive (true);
```
Comment (lw64) [inline src/Library/demos/Toasts/main.vala:11]: ```button_simple.sensitive = true;```

File: src/Library/demos/Toasts/main.vala [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      action_name = "win.undo",
+      action_target = new GLib.Variant.string (message_id),
+      priority = Adw.ToastPriority.HIGH,
+    };
+    overlay.add_toast (advanced_toast);
+  });
+
+  var action_console = new SimpleAction ("undo", new GLib.VariantType ("s"));
+  action_console.activate.connect ((self, target) => {
+      var val = target.get_string ();
```
Comment (lw64) [inline src/Library/demos/Toasts/main.vala:31]: here the type should be also written out, and not just ```var```.

**Review by lw64 — CHANGES_REQUESTED** [review]

**Comment by lw64** [comment]
About the missing Application instance, I think a SimpleActionGroup could be used instead. It should be probably provided by the previewer in the ```workbench``` interface.

Also I think the websocket demo could be object oriented, so that you have a ```WorkbenchWebSocketClient``` or similar that handles the libsoup stuff.

File: src/Library/demos/Toasts/main.vala (hunk 1 above)
Comment (benpocalypse) [reply src/Library/demos/Toasts/main.vala:19]: This is actually a `string` although I could change it to an `int` and then use an `int.parse ()` on line 42, but that seems a tad redundant. Ok to just change it to `string` on line 19 here?

File: src/Library/demos/Toasts/main.vala [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      stdout.printf ("undo" + val);
```
[the remark this answers is not in the document]
Comment (benpocalypse) [reply src/Library/demos/Toasts/main.vala:32]: Indeed there is. Good catch.

File: src/Library/demos/Toasts/main.vala (hunk 1 above)
Comment (lw64) [reply src/Library/demos/Toasts/main.vala:19]: Yeah. I lookedntonquickly over it :?/)

**Review by sonnyp — COMMENTED** [review]
the Adw.Application is now exposed to `workbench.application` thanks to @lw64 https://github.com/sonnyp/Workbench/pull/102

**Comment by sonnyp** [comment]
Also - please feel free to add yourself to the list of contributors in `src/about.js`

**Comment by benpocalypse** [comment]
Yay!
Thanks...I'll get to the changes today.

**Comment by benpocalypse** [comment]
I updated my branch to make use of the newly exposed `application` but the end effect still isn't what is desired. The notifications demo exposes an assertion that fails...not sure what the problem is. Feel free to bonk me on the head.

**Comment by lw64** [comment]
The notification demo is not working currently. Can you split it into a separate commit, that can be later merged, when notifications will work? The erst can be merged I think.

**Comment by benpocalypse** [comment]
> The notification demo is not working currently. Can you split it into a separate commit, that can be later merged, when notifications will work? The rest can be merged I think.

Sure. I'll back out the notifications demo and then we'll see if we can't get this PR merged.

**Review by sonnyp — APPROVED** [review]
Thank you !

I have filed a bug about the issue with actions https://github.com/sonnyp/Workbench/issues/110

**Pull request merged**